py4j-python/src/py4j/java_gateway.py

```diff
@@ -2451,6 +2451,12 @@ def _call_proxy(self, obj_id, input):
             else:
                 java_exception = traceback.format_exc()
 
+            if not isinstance(return_value, JavaObject) and self.gateway_client.converters:
```

Comment:
this code addition is inside the `except Exception` block and should instead be placed before line 2443 (`return proto.RETURN_MESSAGE + proto.SUCCESS +\`